### DISCUSSION TITLE
tycho-surefire-plugin recommended setup

### DISCUSSION THREAD

**Discussion by tivervac** [eclipse-tycho/tycho#2465] [Q&A] [answered]
[The documentation](https://tycho.eclipseprojects.io/doc/master/TestingBundles.html#tycho-surefire-plugin) mentions that using `eclipse-test-plugin` with tycho-surefire-plugin is not recommended for new designs. However, it doesn't mention what _is_ recommended for new designs. Is it just the second option (`eclipse-plugin` packaging with an extra `plugin-test` goal)?

If so, why is this approach preferred?

**Comment by laeubi** [comment, marked as answer]
>  Is it just the second option (eclipse-plugin packaging with an extra plugin-test goal)?

Yes.

> If so, why is this approach preferred?

It does not require a special packaging-type and you can mix surefire with plugin tests / integrationtests as required.

**Reply by tivervac** [reply]
Thanks for the clarification. How would you go about mixing those kinds of tests? Just by pointing tycho-surefire and maven-surefire at different test-source folders?

**Reply by laeubi** [reply]
This section is sadly missing in the documentation (so if you want to provide a demo+documentation PR that would be great), but basically it is a matter of define two "namespaces" so you can define a filter see the following integration test:

https://github.com/eclipse-tycho/tycho/blob/master/tycho-its/projects/surefire.combinedtests/bundle.test/pom.xml

**Reply by tivervac** [reply]
I'll look into providing a documentation PR in the near future
